cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectAssemblyOperatorTest.java

```diff
@@ -433,8 +481,8 @@ public void testReconcile(TestContext context) {
                 asList(KafkaConnectCluster.fromConfigMap(baz).generateDeployment())
         );
 
-        Set<String> createdOrUpdated = new HashSet<>();
-        Set<String> deleted = new HashSet<>();
+        Set<String> createdOrUpdated = new CopyOnWriteArraySet<>();
```

Comment:
I was getting erratic behaviour with this test here: https://github.com/kyguy/strimzi/blob/224cd1101380f6c6f3b6dbf054f2e19315c42645/cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectS2IAssemblyOperatorTest.java#L662 sometimes failing with the assertion error [bar, foo] != [foo] so I thought it might be due to bar either being not being added to the set on the time of the assertion test or deleted prematurely. I noticed that https://github.com/kyguy/strimzi/blob/224cd1101380f6c6f3b6dbf054f2e19315c42645/cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorTest.java#L676-L677 used CopyOnWriteArraySet so I copied it over and the test started passing consistently. Then I copied it over to KafkaConnectAssemblyOperatorTest for symmetry, although the assertion test here never failed.